### PR TITLE
vulkaninfo: Rewrite instance creation

### LINKED ISSUE #100: vulkaninfo should load all instance-level extensions
Due to vulkan 3.33 "Extension Dependencies":

> Any device extension that has an instance extension dependency that is not enabled by vkCreateInstance is considered to be unsupported, hence it must not be returned by vkEnumerateDeviceExtensionProperties for any VkPhysicalDevice child of the instance.

Now, the goal of vulkaninfo is obviously to list all available device-level extensions. However, because of this paragraph in the vulkan spec, any device-level extension that depends on missing instance-level extensions will be censored from the output of vulkaninfo.

I propose changing vulkaninfo in such a way that it loads all available instance-level extensions. (Not sure if we need to be careful on systems with multiple ICDs? Is it possible to load a VkInstance extension that is compatible with one VkPhysicalDevice but not another?)

### PR DESCRIPTION
**Pull request by krOoze**
- localize `VkPhysicalDeviceSurfaceInfo2KHR` (and remove from instance creation)
- <s>Remove debug callback. We are not even enabling any layers...</s>
- Enable all supported instance extensions. Close #100.
- Prettify instance command loader
- other esthetics

### CODE REVIEW

File: vulkaninfo/vulkaninfo.c [hunk 1]
```diff
@@ -834,117 +853,28 @@ static void AppCreateInstance(struct AppInstance *inst) {
...
-
-    VkInstanceCreateInfo inst_info = {.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
-                                      .pNext = NULL,
-                                      .pApplicationInfo = &app_info,
-                                      .enabledLayerCount = 0,
-                                      .ppEnabledLayerNames = NULL,
-                                      .enabledExtensionCount = inst->inst_extensions_count,
-                                      .ppEnabledExtensionNames = inst->inst_extensions};
-
-    VkDebugReportCallbackCreateInfoEXT dbg_info;
```
[the remark this answers is not in the document]
Comment (krOoze) [reply vulkaninfo/vulkaninfo.c]: What kind of driver messages do we want to show in `vulkaninfo` log?
Besides we are not even creating regular debug callback after this, so..

File: vulkaninfo/vulkaninfo.c [hunk 2]
```diff
@@ -818,14 +786,65 @@ bool CheckForJsonOption(const char *arg) {
...
+
+            for (uint32_t ext_i = 0; ext_i < inst->global_extension_count; ++ext_i) {
+                const char *g_ext_name = inst->global_extensions[ext_i].extensionName;
+                const bool already_provided = (strcmp(l_ext_name, g_ext_name) == 0);
+
+                if (!already_provided) {
+                    ++inst->inst_extensions_count;
+                    inst->inst_extensions = realloc((void *)inst->inst_extensions, sizeof(char *) * inst->inst_extensions_count);
+                    if (!inst->inst_extensions) ERR_EXIT(VK_ERROR_OUT_OF_HOST_MEMORY);
+                    inst->inst_extensions[inst->inst_extensions_count - 1] = l_ext_name;
```
[the remark this answers is not in the document]
Comment (krOoze) [reply vulkaninfo/vulkaninfo.c]: Damn, right.
That will actually make the code simpler. We are not enabling any extensions...

File: vulkaninfo/vulkaninfo.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyk-lunarg) [reply vulkaninfo/vulkaninfo.c]: @lenny-lunarg, can you shed some light on this?

[the remark this answers is not in the document]
Comment (jeremyk-lunarg) [reply vulkaninfo/vulkaninfo.c]: While we decide if we want to leave debug info in instance creation or not, please split this into 2 PRs:

1. New method of including instance extensions
2. Removing debug info from instance creation

[the remark this answers is not in the document]
Comment (krOoze) [reply vulkaninfo/vulkaninfo.c]: done

**Pull request merged**